framework/src/actions/AutoCheckpointAction.C

```diff
@@ -25,46 +33,45 @@ AutoCheckpointAction::AutoCheckpointAction(const InputParameters & params) : Act
 void
 AutoCheckpointAction::act()
 {
-  // if there's already a checkpoint object, we don't need to worry about creating a new
-  // checkpoint
+  // if there's already a checkpoint object, we don't need to worry about creating a new one
   const auto checkpoints = _app.getOutputWarehouse().getOutputs<Checkpoint>();
   const auto num_checkpoints = checkpoints.size();
 
+  const bool shortcut_syntax = getParam<bool>("checkpoint");
+
   if (num_checkpoints > 1)
-  {
-    // Get most recently added Checkpoint object and error
-    mooseError("Multiple checkpoints are not allowed. Check the input to ensure there "
-               "is only one Checkpoint defined in the 'Outputs' block, including the "
-               "shortcut syntax 'Outputs/checkpoint=true'.");
-  }
+    checkpoints[0]->mooseError("Multiple Checkpoint objects are not allowed and there is more than "
+                               "one Checkpoint defined in the 'Outputs' block.");
+  if (num_checkpoints == 1 && shortcut_syntax)
+    paramError("checkpoint",
+               "Shortcut checkpoint syntax cannot be used with another Checkpoint object in the "
+               "'Outputs' block");
 
-  // We don't want to set up automatic checkpoints if we are not in the master app
-  else if (_app.isUltimateMaster())
+  if (num_checkpoints == 0)
   {
-    if (num_checkpoints == 0)
-    {
-      // If there isn't an existing checkpoint, init a new one
-      auto cp_params = _factory.getValidParams("Checkpoint");
-      cp_params.setParameters("checkpoint_type", CheckpointType::SYSTEM_CREATED);
-      cp_params.set<bool>("_built_by_moose") = true;
-      _problem->addOutput("Checkpoint", "checkpoint", cp_params);
-    }
+    // If there isn't an existing checkpoint, init a new one
+    auto cp_params = _factory.getValidParams("Checkpoint");
 
-    else // num_checkpoints == 1
-    {
-      // Use the existing Checkpoint object, since we only need to/should make one object the
-      // autosave
-      checkpoints[0]->setAutosaveFlag(CheckpointType::USER_CREATED);
-    }
+    cp_params.set<bool>("_built_by_moose") = true;
```

Comment:
this is just an aside as this is old code - but you can actually query the action system for the current task name. that is, in the checkpoint object you can see if the current task is `auto_checkpoint_action` and by that determine whether or not this is the automatic one.